content-api/content-actors/src/main/scala/org/sunbird/content/upload/mgr/UploadManager.scala

```diff
@@ -26,13 +30,19 @@ object UploadManager {
 		val identifier: String = node.getIdentifier
 		val fileUrl: String = request.getRequest.getOrDefault("fileUrl", "").asInstanceOf[String]
 		val file = request.getRequest.get("file").asInstanceOf[File]
+		val reqFilePath: String = request.getRequest.getOrDefault("filePath", "").asInstanceOf[String].replaceAll("^/+|/+$", "")
+		val filePath = if(StringUtils.isBlank(reqFilePath)) None else Option(reqFilePath)
 		val mimeType = node.getMetadata().getOrDefault("mimeType", "").asInstanceOf[String]
 		val contentType = node.getMetadata.getOrDefault("contentType", "").asInstanceOf[String]
 		val mediaType = node.getMetadata.getOrDefault("mediaType", "").asInstanceOf[String]
 		val mgr = MimeTypeManagerFactory.getManager(contentType, mimeType)
-		val uploadFuture: Future[Map[String, AnyRef]] = if (StringUtils.isNotBlank(fileUrl)) mgr.upload(identifier, node, fileUrl) else mgr.upload(identifier, node, file)
+		val uploadFuture: Future[Map[String, AnyRef]] = if (StringUtils.isNotBlank(fileUrl)) mgr.upload(identifier, node, fileUrl, filePath) else mgr.upload(identifier, node, file, filePath)
 		uploadFuture.map(result => {
-			updateNode(request, node.getIdentifier, mediaType, contentType, result)
+			val newResult = new util.HashMap[String, AnyRef](result)
```

Comment:
Why to create new result object. We can put artifactBasePath in `result` only.
Instead of:

> val newResult = new util.HashMap[String, AnyRef](result)
if(StringUtils.isNotBlank(filePath.getOrElse(""))){
				newResult.put(ContentConstants.ARTIFACT_BASE_PATH, filePath.get)
			}
We can use:

> if(StringUtils.isNotBlank(filePath.getOrElse("")))
				result.put(ContentConstants.ARTIFACT_BASE_PATH, filePath.get)

Reply:
Also remove `/` from start and end.

Reply:
the result object is immutable scalamap i can't add value into it . That's way i created new result object.
I already removed / from start and end in line no. 33 where i am getting filepath from request.